### PR TITLE
Added Partial code Insertion with replacement.

### PR DESCRIPTION
**Pull request by AbhijeetDash**
1. Added LocalCache class to keep track of old code. - used in case of a unaccepted change.
2. Using fuzzyball to fuzzy match the replace block in file content.
3. Separated default prompt into a json format.
4. Added complete cases - complete generation or update to existing code.
5. Restructured some classes and added new functions.

### CODE REVIEW

File: core/helpers/help.code.js
Comment (oi-overide) [inline core/helpers/help.code.js:1]: Please remove the console.log calls to print the old code, new code and block index.

File: core/helpers/help.code.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+     * 
+     * @param {string} fileContent - The content of the file.
+     * @param {Array} codeBlocks - The array of find-and-replace blocks in the format [{ find: "entire old block", replace: "entire new block" }].
+     * @returns {string} - The updated file content.
+     */
+    applyMultipleCodeDiffs(fileContent, codeBlocks) {
+        for (let { find: oldBlock, replace: newBlock } of codeBlocks) {
+            // Step 1: FuzzyFind the entire oldBlock.
+            const oldBlockIndex = this.findFuzzyMatchIndex(fileContent, oldBlock);
+            console.log("OLD CODE: ", oldBlock, "\n");
```
Comment (oi-overide) [inline core/helpers/help.code.js]: Here

File: core/helpers/help.code.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+            const score = fuzzball.ratio(cleanedOldBlock, cleanedChunk);
+
+            // If we get a higher score, update the best match and store the index
+            if (score > bestScore) {
+                bestScore = score;
+                bestMatch = index;
+            }
+        });
+
+        console.log('Best match score:', bestScore);
```
Comment (oi-overide) [inline core/helpers/help.code.js]: Also remove this.

File: core/helpers/help.code.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+     * @param {number} matchIndex - The index of the old block in the file content.
+     * @returns {string} - The updated file content.
+     */
+    replaceOldCodeWithNew(fileContent, oldBlock, newBlock, matchIndex) {
+        if (matchIndex !== -1) {
+            const updatedContent = fileContent.split('\n\n').map((chunk, index) => {
+                return index === matchIndex ? newBlock : chunk;
+            }).join('\n\n');
+
+            console.log("Old Block Replaced: ", oldBlock);
```
Comment (oi-overide) [inline core/helpers/help.code.js]: Also this.

File: core/network/network.js [hunk 4]
```diff
@@ -18,6 +18,8 @@ class Network {
   async doRequest(requestData) {
     const { activeServiceDetails } = requestData;
 
+    console.log("MAKING NETWORK CALL");
```
Comment (oi-overide) [inline core/network/network.js]: Also this.

File: package.json [hunk 5]
```diff
@@ -16,7 +16,10 @@
     "axios": "^1.7.7",
     "chokidar": "^4.0.0",
     "commander": "^12.1.0",
+    "diff": "^7.0.0",
     "dotenv": "^16.4.5",
+    "fuzzball": "^2.1.3",
+    "fuzzy": "^0.1.3",
```
Comment (oi-overide) [inline package.json]: Not used

File: package.json [hunk 6]
```diff
@@ -16,7 +16,10 @@
     "axios": "^1.7.7",
     "chokidar": "^4.0.0",
     "commander": "^12.1.0",
+    "diff": "^7.0.0",
```
Comment (oi-overide) [inline package.json]: not used

**Review by ghost — CHANGES_REQUESTED** [review]
Please make the required changes.

**Review by ghost — APPROVED** [review]

**Pull request merged**